### DISCUSSION TITLE
Text + image as label of an horizontal bar chart in Plot

### DISCUSSION THREAD

**Discussion by espinielli** [observablehq/plot#2263] [Q&A] [answered]
I would like to have a y label composed of both text and image in an horizontal bar chart in Plot.
I have an example [here](https://observablehq.com/d/65e66f2c5e90805a) where I would like to add the president image to the right of the relevant name on the y axis.

Any hints at how it could be achieved?

Is there a way to combine marks together (like a Plot.Text and an Plot.Image...if that would be a possible path to a solution)?

I am lost...

Any helps is welcome

**Comment by mbostock** [comment, marked as answer]
The main problem with your notebook is that you haven’t specified the types with your CSV. The default types are strings, so you end up with the _x_ scale being ordinal by default instead of linear. You can fix this by specifying `typed: true` when loading the CSV:

```js
presidents = FileAttachment("us-president-favorability-1.csv").csv({typed: true})
```

That fixed, you can simply add a Plot.image mark if you want an image.

![untitled - 2024-12-23T090441 016](https://github.com/user-attachments/assets/b69a2c35-b5e1-4b63-b9ec-3d1c01c0ba6a)

```js
Plot.plot({
  marginLeft: 170,
  marks: [
    Plot.barX(presidents, {
      y: "Name",
      x: "Very Favorable %"
    }),
    Plot.image(presidents, {
      y: "Name",
      x: "Very Favorable %",
      dx: 12,
      r: 9,
      preserveAspectRatio: "xMidYMin slice",
      src: "Portrait URL"
    })
  ]
})
```

Likewise if you want text, you can add a text mark. If you want to combine marks, take a look at Plot.marks, or just write a function that returns an array of marks.

**Comment by espinielli** [comment]
Ah, I haven't expressed myself clearly enough...
I am aiming at having for example "Abraham Lincoln `<image>`" as the bar label...maybe it is just a matter of pushing the names to the left and the Plot.image's to a negative position?

**Comment by espinielli** [comment]
Ok, I managed to do the following:
![imaged_presidents](https://github.com/user-attachments/assets/2a9b162e-1ebe-478b-ade4-b7a16d13f001)



```js
Plot.plot({
  marginLeft: 170,
  marks: [
    Plot.barX(presidents, {
      y: (d) => d["Name"],
      x: "Very Favorable %"
    }),
    Plot.image(presidents, {
      y: "Name",
      x: 0,
      dx: -12,
      r: 9,
      preserveAspectRatio: "xMidYMin slice",
      src: "Portrait URL"
    }),
    Plot.axisY({
      label: null,
      textAnchor: "end",
      fill: "black",
      dx: -18
    })
  ]
})
```

**Reply by espinielli** [reply]
still need to remove those ticks but progress...
Thank for heading me in the right direction @mbostock

**Reply by espinielli** [reply]
I changed order of axisY to come before images, so as to partially hide the ticks

**Comment by espinielli** [comment]
I *enhanced* (my judgement ;-) it as follows
![untitled (1)](https://github.com/user-attachments/assets/f63f6694-d174-48bd-ba55-0c0be1f83a25)


```js
Plot.plot({
  marginLeft: 170,
  marks: [
    Plot.barX(presidents, {
      y: (d) => d["Name"],
      x: "Very Favorable %"
    }),
    Plot.axisY({
      label: null,
      textAnchor: "end",
      fill: "black",
      tickPadding: 18
    }),
    Plot.image(presidents, {
      y: "Name",
      x: 0,
      dx: -12,
      r: 9,
      preserveAspectRatio: "xMidYMin slice",
      src: "Portrait URL"
    })
  ]
})
```
